plutus-tx-plugin/src/PlutusTx/Compiler/Expr.hs

```diff
@@ -929,6 +931,44 @@ compileExpr e = traceCompilation 2 ("Compiling expr:" GHC.<+> GHC.ppr e) $ do
   caseIntegerName <- lookupGhcName 'Builtins.caseInteger
   listIndexId <- lookupGhcId '(PlutusTx.List.!!)
 
+  let
+    compileMkNil
+      :: (CompilingDefault uni fun m ann)
+      => GHC.Type
+      -> m (PIRTerm uni fun)
+    compileMkNil ty =
+      let
+        tyToUniverse :: GHC.Type -> Maybe (SomeTypeIn' PLC.DefaultUni)
```

Comment:
is there a way to do this with existing `SomeTypeIn`? `SomeTypeIn` won't work because it has existential kind and it's impossible to force that to `Type`...

Reply:
I think you can avoid existentials altogether and do something like

```haskell
withCompiledMkNil :: GHC.Type -> (forall a. PLC.DefaultUni (Esc a) -> m (PIRTerm uni fun))  -> m (PIRTerm uni fun)
```

but I also think that your version is simpler, so who cares.

Reply:
I refuse to use rank-n-types whenever possible